### PR TITLE
Sign hashes in a simple contract

### PR DESCRIPTION
**Pull request by jksf**
This is a simple contract with a single test. There are several issues to be solved but I'd like to have this code as a basis for the discussion.

### CODE REVIEW

File: globals.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  address: string;
+}
+
+declare interface ISignHash extends IContractInstance {
+  sign: (hash: string) => Promise<void>;
+  getSigners: (hash: string) => Promise<string[]>;
+}
+
+declare interface IArtifacts {
+  require: (path: string) => IContract;
```
Comment (biern) [inline globals.d.ts]: I think we can take more advantage of TS here using generic types and literal typing. Please checkout 
https://github.com/biern/truffle-typescript-example/blob/master/contracts/truffle.d.ts

File: contracts/SignHash.sol [hunk 2]
```diff
@@ -0,0 +1,32 @@
+pragma solidity ^0.4.0;
+
+
+contract SignHash {
+
+    //--- Storage
+
+    mapping (bytes32 => address[]) private hashSigners;
```
Comment (biern) [inline contracts/SignHash.sol:8]: I'm wondering, how do you intend to store Signer "identities" to verify (I belive we don't have a word for that, I mean their domain name / github organization)? As instances of another contract?

File: globals.d.ts (hunk 1 above)
Comment (jksf) [reply globals.d.ts]: Thanks! I'm still learning TS, I made changes inspired by your example.

File: contracts/SignHash.sol (hunk 2 above)
Comment (jksf) [reply contracts/SignHash.sol:8]: I see it as a `mapping (address => Signer)` with `struct Signer` containing information about registered identity verification methods.

**Comment by jksf** [comment]
@biern I plan to add Signer registration in another PR. If you like the code please merge.

File: contracts/SignHash.sol (hunk 2 above)
Comment (biern) [reply contracts/SignHash.sol:8]: 👍 I'm thinking we might need to use mappings to store (`verification_source` -> `identity`, eg. `domain`: `signhash.com`) for each signer to make that future proof. But if you have something else in mind that's a discussion for a separate issue.

Comment (jksf) [reply contracts/SignHash.sol:8]: :+1: Yes, this is a good idea, I was thinking about extensibility without redeployment of the contract but its a separate issue.

**Comment by biern** [comment]
Looks good 👍 
BTW, I don't consider putting `I` in front of interfaces a good practice, do you have any strong opinion on that? If not I suggest to drop that.
BTW, I don't consider putting `I` in front of interfaces a good practice, do you have any strong opinion on that? If not I suggest to drop that.

**Comment by jksf** [comment]
I'm confused. TypeScript official guidelines specifies:
`Do not use "I" as a prefix for interface names.`
But TSLint in default configuration requires it. I will adjust TSLint config and drop this prefix. I'm not a huge fan of this convention.

**Comment by biern** [comment]
😁 great, so just merge when you're done

**Pull request merged**